### PR TITLE
 memory: make buffers aligned to CACHE_LINE_SIZE

### PR DESCRIPTION
**Pull request by abonislawski**
As adviced in PR #1557 it provides better memory/cache performance since it aligns with memory IO burst size and cache line size

This PR will use CACHE_LINE_SIZE in CNL, ICL and SUE

### CODE REVIEW

File: src/platform/cannonlake/cannonlake.x.in [hunk 1]
```diff
@@ -443,7 +443,7 @@ SECTIONS
     . = . + HEAP_RUNTIME_SIZE;
     _runtime_heap_end = ABSOLUTE(.);
 
-    . = ALIGN (32);
+    . = ALIGN (XCHAL_DCACHE_LINESIZE);
```
Comment (tlauda) [inline src/platform/cannonlake/cannonlake.x.in]: Please use PLATFORM_DCACHE_ALIGN.

Comment (abonislawski) [reply src/platform/cannonlake/cannonlake.x.in]: @tlauda currently not possible, we cannot include sof/platform.h to linker and the value of PLATFORM_DCACHE_ALIGN can be "sizeof(uint32_t)" which will not work in linker.

If we want to use it then we need to move it to another file and remove sizeof possibility

Comment (mmaka1) [reply src/platform/cannonlake/cannonlake.x.in]: This is platform specific linker script, not a universal one, so I'm fine with some HEAP_BUF_ALIGNMENT define in some common cAVS header file included by the linker script and also in another place followed by some C_ASSERT(0 == (HEAP_BUF_ALIGNMENT % PLATFORM_DCACHE_ALIGN)).

Comment (abonislawski) [reply src/platform/cannonlake/cannonlake.x.in]: @tlauda @mmaka1 added HEAP_BUF_ALIGNMENT with static_assert for it.
Done in cAVS memory.h and memory.c, please check if this is a correct place for them

**Comment by abonislawski** [comment]
> commit message and please comment what this check does in the code.

done

**Comment by abonislawski** [comment]
SOFCI TEST

**Comment by abonislawski** [comment]
No changes, just force-pushed to trigger travis again

**Comment by abonislawski** [comment]
@lgirdwood  please finish review. We need it for #1557 which is needed for #1567 :)

**Pull request merged**